src/api/lib/backend/api.rb

```diff
@@ -15,11 +15,18 @@ def self.write_attributes(project, package, login, xml, comment)
       Backend::Connection.put(path, xml)
     end
 
-    # Returns a file list (from src/api/app/controllers/build/file_controller.rb)
-    def self.file_list(project, repository, arch, package)
+    # Returns a file list of binaries (from src/api/app/controllers/build/file_controller.rb)
+    def self.binary_files_list(project, repository, arch, package)
       Backend::Connection.get("/build/#{CGI.escape(project)}/#{CGI.escape(repository)}/#{CGI.escape(arch)}/#{CGI.escape(package)}").body
     end
 
+    # Returns a file list of the sources for a package
+    def self.file_list(project, package, options = {})
+      path = "/source/#{CGI.escape(project)}/#{CGI.escape(package)}"
```

Comment:
why not using directly `Package.source_path(project, package, nil, opts)` instead of duplicating the code here?

Reply:
@Ana06, because at the end we want to remove all these kind of methods that calculate the URL to connect to the backend, so at the moment we still live with the duplication until we move all these methods to Backend::Api :smile_cat:

Reply:
That's it, no url calculations out of the Backend::Api... whenever we are going to be done with the move of all the Backend::Connection calls to there, then we can refactor and maybe share code between methods that use the same base url...